### ISSUE TITLE
{pathinfo: true} can break JS syntax in built files when the loader query contains */

### ISSUE THREAD

**Issue by mik01aj** [webpack/webpack#1759]
Here's a piece of JS that was emitted by webpack:

	/* WEBPACK VAR INJECTION */(function(global) {module.exports = global["MobileRouter"] = __webpack_require__(/*! -!/Users/mikolajdadela/code/lb-react-frontend/~/babel-loader?{"plugins":["transform-react-jsx"],"ignore":["node_modules/**/*"]}!./mobilePages/MobileRouter.jsx */ 1);
	/* WEBPACK VAR INJECTION */}.call(exports, (function() { return this; }())))

A shortened version of the first line:

	/* WEBPACK ... */( ... __webpack_require__(/* ... ["node_modules/**/*"] ...  */ 1);
                                                                       ^

When I run this code in browser, I get

    Uncaught SyntaxError: Unexpected token *

**Comment by bebraw** [comment]
@mik01aj Can you provide a standalone test case?

**Comment by mik01aj** [comment]
I don't have much time for this, but I guess that...

    require('babel?query={"ignore": ["node_modules/**/*"]}!./somefile.js');

...would be enough.

**Comment by cletusw** [comment]
@bebraw Minimal test case:

webpack.config.js
```javascript
module.exports = {
  entry: {
    app: ['./main.js?*/']
  },
  output: {
    path: __dirname,
    filename: 'bundle.js',
    pathinfo: true
  }
};
```

main.js
```javascript
console.log('hello, world');
```

steps to reproduce
```
npm install webpack
node_modules/.bin/webpack
```

bundle.js (after the initial webpack stuff)
```javascript
/* 0 */
/*!*****************!*\
  !*** multi app ***!
  \*****************/
/***/ function(module, exports, __webpack_require__) {

	module.exports = __webpack_require__(/*! ./main.js?*/ */1);


/***/ },
/* 1 */
/*!********************!*\
  !*** ./main.js?*_/ ***!
  \********************/
/***/ function(module, exports) {

	console.log('howdy');


/***/ }
/******/ ]);
```

Note the syntax error in chunk `0`'s `module.exports` line because the query string prematurely closes the comment. Also note that this construct is correctly escaped in the header comment for chunk `1` as `./main.js?*_/`

**Comment by bebraw** [comment]
Yeah, I can see it's missing escaping. Thanks! That's a definite bug.

**Comment by cletusw** [comment]
@bebraw The escaping is also not happening for `devtool: 'eval'`

webpack.config.js
```javascript
module.exports = {
  entry: './main.js?*/',
  output: {
    path: __dirname,
    filename: 'bundle.js'
  },
  devtool: 'eval'
};
```

main.js
```javascript
console.log('hello, world');
```

steps to reproduce
```
npm install webpack
node_modules/.bin/webpack
```

bundle.js (after the initial webpack stuff)
```javascript
/* 0 */
/***/ function(module, exports) {

	eval("console.log('howdy');\n\n\n/*****************\n ** WEBPACK FOOTER\n ** ./main.js?*/\n ** module id = 0\n ** module chunks = 0\n **/\n//# sourceURL=webpack:///./main.js?");

/***/ }
/******/ ]);
```

The `WEBPACK FOOTER` also is missing escaping, which, when run in the browser, gives `Uncaught SyntaxError: Unexpected token **`

**Comment by cletusw** [comment]
Although in 2.x, the `WEBPACK FOOTER` uses `//` comments, so `devtool: 'eval'` issue doesn't exist there.

**Comment by cletusw** [comment]
Sorry for all the notifications. I've pushed a fix to PR #2916

**Comment by cletusw** [comment]
The `pathinfo: true` issue is still very much present in webpack 3.5.3.
